File: java/com/joshmanisdabomb/lcc/item/HeartContainerItem.java
package com.joshmanisdabomb.lcc.item;

import com.joshmanisdabomb.lcc.data.capability.HeartsCapability;
import com.joshmanisdabomb.lcc.functionality.HeartsFunctionality;
import net.minecraft.entity.LivingEntity;

public class HeartContainerItem extends HeartItem {

    public HeartContainerItem(HeartsFunctionality.HeartType ht, float value, Properties p) {
        super(ht, value, p);
    }

    protected void onHeartEffect(HeartsCapability.CIHearts hearts, LivingEntity entity) {
        this.ht.addMaxHealth(hearts, entity, this.value);
        this.ht.addHealth(hearts, entity, this.value, HeartsFunctionality.TEMPORARY_USUAL_LIMIT);
    }

    protected boolean canUse(HeartsCapability.CIHearts hearts, LivingEntity entity) {
        return !this.ht.isFullMaxHealth(hearts, entity);
    }

}

File: kotlin/com/joshmanisdabomb/lcc/item/HeartContainerItem.kt
package com.joshmanisdabomb.lcc.item

import com.joshmanisdabomb.lcc.concepts.heart.HeartType
import net.minecraft.entity.LivingEntity

class HeartContainerItem(heart: HeartType, value: Float, settings: Settings) : HeartItem(heart, value, settings) {

    override fun canUseHeart(user: LivingEntity) = heart.getMaxHealth(user) < heart.getDefaultMaxLimit(user)

    override fun useHeart(user: LivingEntity) = heart.addMaxHealth(user, value)

}